spec/controllers/foods_controller_spec.rb

```diff
@@ -1,159 +1,191 @@
+          expect(response).to(redirect_to(new_recipe_ingredient_url(recipe, ingredient: { food_id: Food.last.to_param })))
+        end
+      end
+
+      context "with invalid params" do
+        it "assigns a newly created but unsaved food as @food" do
+          post :create, food: { name: "" }
+          expect(assigns(:food)).to be_a_new(Food)
+        end
+
+        it 're-renders the "new" template' do
+          post :create, food: { name: "" }
+          expect(response).to render_template("new")
+        end
       end
     end
 
-    context "with invalid params" do
-      it "assigns the food as @food" do
-        food = Food.create! valid_attributes
-        put :update, {:id => food.to_param, :food => invalid_attributes}, valid_session
-        expect(assigns(:food)).to eq(food)
+    describe 'PUT #update' do
+      context "with valid params" do
+        let(:new_attributes) { { name: "New Name" } }
+
+        it "raises an error if a non-user food is posted" do
+          food = FactoryGirl.create(:food)
+          expect { put :update, id: food.to_param, food: new_attributes }.
+            to raise_error(ActiveRecord::RecordNotFound)
+        end
+
+        it "updates the requested food" do
+          food = FactoryGirl.create(:food, user: @current_user)
+          put :update, id: food.to_param, food: new_attributes
+          expect(food.reload.name).to eq("New Name")
+        end
+
+        it "assigns the requested food as @food" do
+          food = FactoryGirl.create(:food, user: @current_user)
+          put :update, id: food.to_param, food: valid_attributes
+          expect(assigns(:food)).to eq(food)
+        end
+
+        it "redirects to the food" do
+          food = FactoryGirl.create(:food, user: @current_user)
+          put :update, id: food.to_param, food: valid_attributes
+          expect(response).to redirect_to(foods_url)
+        end
       end
 
-      it "re-renders the 'edit' template" do
-        food = Food.create! valid_attributes
-        put :update, {:id => food.to_param, :food => invalid_attributes}, valid_session
-        expect(response).to render_template("edit")
+      context "with invalid params" do
+        it "assigns the food as @food" do
+          food = FactoryGirl.create(:food, user: @current_user)
+          put :update, id: food.to_param, food: { name: "" }
+          expect(assigns(:food)).to eq(food)
+        end
+
+        it 're-renders the "edit" template' do
+          food = FactoryGirl.create(:food, user: @current_user)
+          put :update, id: food.to_param, food: { name: "" }
+          expect(response).to render_template("edit")
+        end
       end
     end
-  end
 
-  describe "DELETE #destroy" do
-    it "destroys the requested food" do
-      food = Food.create! valid_attributes
-      expect {
-        delete :destroy, {:id => food.to_param}, valid_session
-      }.to change(Food, :count).by(-1)
-    end
+    describe 'DELETE #destroy' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.